apps/platform/src/components/members/membersTable/index.tsx

```diff
@@ -0,0 +1,183 @@
+'use client'
+
+import * as React from 'react'
+import { useAtom, useAtomValue } from 'jotai'
+import { ChevronLeft, ChevronRight, ChevronsLeft, ChevronsRight } from 'lucide-react'
+import dayjs from 'dayjs'
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow
+} from '@/components/ui/table'
+import { membersOfWorkspaceAtom, selectedWorkspaceAtom } from '@/store'
+import { useHttp } from '@/hooks/use-http'
+import ControllerInstance from '@/lib/controller-instance'
+import AvatarComponent from '@/components/common/avatar'
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from '@/components/ui/select'
+import { Button } from '@/components/ui/button'
+
+export interface Payment {
+  id: string
+  amount: number
+  status: 'pending' | 'processing' | 'success' | 'failed'
+  email: string
+}
+
+export interface Member {
+  id: string
+  user: {
+    name: string
+    email: string
+    profilePicture: string
+  }
+  joiningDate: string
+  roles: string[]
+}
+
+export default function MembersTable(): React.JSX.Element {
+  const [members, setMembers] = useAtom(membersOfWorkspaceAtom)
+  const currentWorkspace = useAtomValue(selectedWorkspaceAtom)
+  const [currentPage, setCurrentPage] = React.useState(1)
+  const [rowsPerPage, setRowsPerPage] = React.useState(10)
+
+  const getAllMembers = useHttp(() =>
+    ControllerInstance.getInstance().workspaceMembershipController.getMembers(
+      {
+        workspaceSlug: currentWorkspace!.slug
+      },
+      {}
+    )
+  )
+
+  // Calculate pagination values
+  const totalMembers = members.length

```

Comment:
You wont need to calculate these. the response will have this in the pagination metadata